src/core/server/saved_objects/migrationsv2/integration_tests/corrupt_outdated_docs.test.ts

```diff
@@ -104,20 +106,39 @@ describe('migration v2 with corrupt saved object documents', () => {
         '7.14.0': (doc) => doc,
       },
     });
-    try {
-      await root.start();
-    } catch (err) {
-      const corruptFooSOs = /foo:/g;
-      const corruptBarSOs = /bar:/g;
-      const corruptBazSOs = /baz:/g;
-      expect(
-        [
-          ...err.message.matchAll(corruptFooSOs),
-          ...err.message.matchAll(corruptBarSOs),
-          ...err.message.matchAll(corruptBazSOs),
-        ].length
-      ).toEqual(16);
-    }
+    await expect(root.start()).rejects.toThrowError();
```

Comment:
Why don't use the old approach? Reading from the logs adds flakiness (logging nondeterminism, the test depends on the logging format, etc.)

Reply:
Ok, no problem. I'll change it back to read from the error message itself.